test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/cdc/CDCE2EIT.java

```diff
@@ -0,0 +1,230 @@
+                getDatabaseType());
+        PipelineDataSourceWrapper sourceDataSource = new PipelineDataSourceWrapper(generateShardingSphereDataSourceFromProxy(), getDatabaseType());
+        StandardPipelineTableMetaDataLoader metaDataLoader = new StandardPipelineTableMetaDataLoader(targetDataSource);
+        PipelineTableMetaData tableMetaData = metaDataLoader.getTableMetaData(PipelineBaseE2EIT.SCHEMA_NAME, "t_order");
+        PipelineColumnMetaData primaryKeyMetaData = tableMetaData.getColumnMetaData(tableMetaData.getPrimaryKeyColumns().get(0));
+        ConsistencyCheckJobItemProgressContext progressContext = new ConsistencyCheckJobItemProgressContext("", 0);
+        SingleTableInventoryDataConsistencyChecker checker = new SingleTableInventoryDataConsistencyChecker("", sourceDataSource, targetDataSource, schemaTableName, schemaTableName,
+                primaryKeyMetaData, metaDataLoader, null, progressContext);
+        DataConsistencyCheckResult checkResult = checker.check(new DataMatchDataConsistencyCalculateAlgorithm());
+        System.out.println(checkResult);
+    }
+    
+    private void registerStorageUnit() throws SQLException {
+        String registerStorageUnitTemplate = REGISTER_STORAGE_UNIT_SQL.replace("${user}", getUsername())
+                .replace("${password}", getPassword())
+                .replace("${ds0}", appendExtraParam(getActualJdbcUrlTemplate(DS_0, true)))
+                .replace("${ds1}", appendExtraParam(getActualJdbcUrlTemplate(DS_1, true)));
+        addResource(registerStorageUnitTemplate);
+    }
+    
+    private void createOrderTableRule() throws SQLException {
+        proxyExecuteWithLog(CREATE_SHARDING_RULE_SQL, 2);
+    }
+    
+    private void initSchemaAndTable(final Connection connection) throws SQLException {
+        if (getDatabaseType().isSchemaAvailable()) {
+            String sql = String.format("CREATE SCHEMA %s", PipelineBaseE2EIT.SCHEMA_NAME);
+            log.info("create schema sql: {}", sql);
+            connection.createStatement().execute(sql);
+        }
+        String sql = getExtraSQLCommand().getCreateTableOrder(getSourceTableOrderName());
+        log.info("create table sql: {}", sql);
+        connection.createStatement().execute(sql);
+    }
+    
+    private void startCDCClient() {
+        ImportDataSourceParameter importDataSourceParam = new ImportDataSourceParameter(appendExtraParam(getActualJdbcUrlTemplate(DS_4, false, 0)), getUsername(), getPassword());
+        StartCDCClientParameter parameter = new StartCDCClientParameter(importDataSourceParam);
+        parameter.setAddress("localhost");
+        parameter.setPort(getContainerComposer().getProxyCDCPort());
+        parameter.setUsername(ProxyContainerConstants.USERNAME);
+        parameter.setPassword(ProxyContainerConstants.PASSWORD);
+        parameter.setDatabase("sharding_db");
+        // TODO add full=false test case later
+        parameter.setFull(true);
+        String schema = "";
```

Comment:
Is empty default schema required? Could it be null?

Reply:
Required, protobuf Message not allow to use `null`

Reply:
OK